assembly/src/ast/mod.rs

```diff
@@ -64,13 +71,22 @@ impl ProgramAst {
     /// Returns a new [ProgramAst].
     ///
     /// A program consist of a body and a set of internal (i.e., not exported) procedures.
-    pub fn new(local_procs: Vec<ProcedureAst>, body: Vec<Node>) -> Result<Self, ParsingError> {
+    pub fn new(
+        imports: BTreeMap<String, LibraryPath>,
+        local_procs: Vec<ProcedureAst>,
+        body: Vec<Node>,
+    ) -> Result<Self, ParsingError> {
```

Comment:
minor nit: let's change the order of parameters here to be:
```Rust
pub fn new(
    body: Vec<Node>,
    local_procs: Vec<ProcedureAst>,
    imports: BTreeMap<String, LibraryPath>,
) -> Result<Self, ParsingError>
```
And also make a similar re-ordering in the struct definition.